File: ecos-records/src/main/java/ru/citeck/ecos/records3/source/common/AttValueCtx.java
package ru.citeck.ecos.records3.source.common;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import ru.citeck.ecos.commons.data.DataValue;
import ru.citeck.ecos.commons.data.ObjectData;
import ru.citeck.ecos.records3.RecordRef;

import java.util.Collection;
import java.util.Map;

public interface AttValueCtx {

    @NotNull
    RecordRef getRef() throws Exception;

    @NotNull
    String getLocalId() throws Exception;

    @NotNull
    DataValue getAtt(@NotNull String attribute) throws Exception;

    @Nullable
    <T> T getAtts(@NotNull Class<T> type) throws Exception;

    @NotNull
    ObjectData getAtts(@NotNull Map<String, String> attributes) throws Exception;

    @NotNull
    ObjectData getAtts(@NotNull Collection<String> attributes) throws Exception;
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/value/AttValueCtx.kt
package ru.citeck.ecos.records3.record.atts.value

import ru.citeck.ecos.commons.data.DataValue
import ru.citeck.ecos.commons.data.ObjectData
import ru.citeck.ecos.records2.RecordRef
import ru.citeck.ecos.webapp.api.entity.EntityRef

interface AttValueCtx {

    /**
     * Get internal raw value from this context.
     * Returned value should be ready to call recordsService.getAtts(value, atts)
     */
    fun getValue(): Any

    fun getRef(): RecordRef

    fun getRawRef(): EntityRef

    fun getLocalId(): String

    fun getAtt(attribute: String): DataValue

    fun getAtts(attributes: Map<String, *>): ObjectData

    fun getAtts(attributes: Collection<String>): ObjectData

    fun <T : Any> getAtts(attributes: Class<T>): T
}
